### PR TITLE
Aligned read-applications returned data to read-application schema

### PR DESCRIPTION
**Pull request by amosciatti**
Fixes pitchfork/sunrise/issues/2629

Aligned read-applications returned data to read-application adding following missing details:

last_modified_by
description
specification_type
criticality_level (already declared in kio-api.yaml but not returned currently)
created
created_by
publicly_accessible

### CODE REVIEW

**Comment by amosciatti** [comment]
PR live in stups-test and correctly working
https://kio.stups-test.zalan.do/apps

this is one of the returned apps as example, and is exactly the same as querying one single app via `/apps/sunrise`:

```
{
  "last_modified": "2022-03-04T08:11:51.981Z",
  "description": "Sunrise is the new iteration of the Developer Console. It is a platform that is providing a centralized way for Builders to access and use all the tooling, information, and services required for their daily work; presenting this content in a beautiful, accessible, and consistent way so that it is both discoverable and easy to use.",
  "documentation_url": "https://sunrise.docs.zalando.net/",
  "specification_url": "https://github.bus.zalan.do/pitchfork/issues/issues/",
  "publicly_accessible": false,
  "service_url": "https://sunrise.zalando.net",
  "name": "Sunrise, The Developer Console as a platform",
  "criticality_level": 2,
  "created": "2020-07-23T12:35:43.812Z",
  "scm_url": "https://github.bus.zalan.do/pitchfork/sunrise",
  "last_modified_by": "amosciatti",
  "active": true,
  "id": "sunrise",
  "subtitle": "Implementing the Developer Console as a platform.",
  "team_id": "pitchfork",
  "incident_contact": "Developer Productivity",
  "created_by": "gnagarajan",
  "specification_type": null,
  "support_url": "https://github.bus.zalan.do/pitchfork/issues/issues/new/choose"
}
```

File: resources/api/kio-api.yaml [hunk 1]
```diff
@@ -580,12 +605,6 @@ definitions:
         minimum: 1 # Tier-1
         maximum: 4 # not relevant
         example: 2
-      required_approvers:
-        type: integer
-        minimum: 1
-        description: DEPRECATED! WILL BE REMOVED IN NEXT RELASE! Minimum number of approvers needed for a version. Used by fullstop.
-        example: 2
-        default: 2
```
Comment (tkrop) [inline resources/api/kio-api.yaml]: @amosciatti Removing this deprecated features imposes the low risk that we break some consumers not conforming to the latest API doc. I'm wondering whether this is an unnecessary risk we should not take now. We have a high number of other deprecated features that are not removed yet and we are not prepared to support this deprecation now. I think removing this is currently not worth the risk.

**Comment by amosciatti** [comment]
new PR live in stups-test and correctly working

**Comment by amosciatti** [comment]
👍

**Comment by tkrop** [comment]
👍

**Comment by kgalli** [comment]
As I'm missing context: I think the change itself is fine but might have an unwanted side effect. As we are returning significant amount of more data we might slow down requests. Especially as the description field of the application is exposed. I do not think people store strings of whole book size inside but I also do not know. If calls can be (and typically are) paginated I think that should all be fine. Else please evaluate if that might be an issue.

**Comment by amosciatti** [comment]
@kgalli the difference of replying time is quite small here (you can test it trying to query production cluster (old api) and test cluster (new one).
Regarding amount of data, old API return 2 Mb while new one 3.2.
Creating pagination could be an improvement, but would require several effort (apart if you already know how to implement it quickly in Clojure) for a problem that in my opinion is not currently visible.
On the other side what currently happen is that Sunrise query first once for all the apps and then send 4000 queries to get the rest of the data in few time and this, in my opinion is causing more issue then the data amount of data transmitted
Another possibility (as was my initial change) could be to create an ad-hoc endpoint, to avoid problems to other users utilising Kio 
What do you think?

**Comment by tkrop** [comment]
My two cents on this:

* Adding pagination would break existing clients, so it is no option.
* Removing the description would not be sufficient for the Sunrise import process.
* We could utilize the `Prefer`-header or an optional query parameter to enable selected access to the additional fields, however I think:
  * The additional data loaded by the database and provided to the server is probably insignificant base on how the data is stored.
  * Since the amount of data transferred from the server to the client is not even twice as much I think the change is not significant. We can safe significantly more transfer volume by enabling encoding `Accept-encoding: gzip`.@amosciatti have you already activated this?

**Comment by amosciatti** [comment]
> We can safe significantly more transfer volume by enabling encoding Accept-encoding: gzip.@amosciatti have you already activated this?

No @tkrop , I will implement it in my next PR when Kio PR is merged; thanks fro your hint

**Comment by kgalli** [comment]
👍

**Comment by kgalli** [comment]
Let's merge this one and tackle the topics discussed in follow up PRs.

**Pull request merged**